### ISSUE TITLE
Web socket connections started by web workers are not proxied

### ISSUE THREAD

**Issue by kvindascr** [DevExpress/testcafe#7128]
### What is your Scenario?

Hello TestCafe team.

We have a situation where WebSocket connections over web workers are not being proxied.

We need to pass authentication cookies to a web socket connection but the connection is started by a web worker, in that specific case, we can see that the websocket connection URL is not proxied (see attached screenshot), and the auth cookies are not being attached to the request.

### What is the Current behavior?

Secure web socket connections initiated by a web worker are not proxied.
Authentication cookies added during the test execution, are missing from the web socket connection.

### What is the Expected behavior?

Websockets connection initiated by a web worker should be proxied.
Authentication cookies should be added to web socket connections initiated by a web worker.

### What is your public website URL? (or attach your complete example)

https://github.com/kvindascr/test-cafe-cookie-issue

Follow the readme instructions to setup and run the example.

### What is your TestCafe test code?

import { debug } from '../testcafe/log';
import { ReactSelector } from 'testcafe-react-selectors';
import {ClientFunction, Selector} from 'testcafe';

const STABLE_TEST_ROOM = 'stable-baa';
const url1 = 'google.com';
const url2 = 'news.ycombinator.com';
const url3 = 'linear.app';

const setCookiesBtn = ReactSelector('AppContainer');
const callWSBtn = ReactSelector('AppContainer').withProps({ clientId: this.clientId }).findReact('BrowserAppContainer');

export const getPageHTML = ClientFunction(() => {
    return document.documentElement.outerHTML;
});

fixture('TestFixture').beforeEach(async (t) => {
    debug('starting fixture');
});

test('add browser and test core actions', async (t) => {
    await t.navigateTo('https://local.myapp.app/');
    // await t.debug();
    await t.wait(5000);
    await t.click('#setCookiesBtn');
    await t.click('#callSocketBtn');
    await t.wait(2000);
    // await t.debug();
    await t.expect(Selector('#result').textContent).contains('Cookies found');
});

### Your complete test report

```
$ node ./run-test-cafe.js
 Running tests in:
 - Chrome 103.0.5060.53 / macOS 10.15.7

 TestFixture
starting fixture
 ✖ add browser and test core actions

   1) AssertionError: expected 'ERROR Cookies NOT found' to include 'Cookies found'

      Browser: Chrome 103.0.5060.53 / macOS 10.15.7

         24 |    await t.wait(5000);
         25 |    await t.click('#setCookiesBtn');
         26 |    await t.click('#callSocketBtn');
         27 |    await t.wait(2000);
         28 |    // await t.debug();
       > 29 |    await t.expect(Selector('#result').textContent).contains('Cookies found');
         30 |});
         31 |

         at <anonymous> (/Users/kvindas/Development/js-projects/switchboard/test-cafe-cookie-issue/testing/e2e/tests/CookiesTest.test.ts:29:53)
         at fulfilled (/Users/kvindas/Development/js-projects/switchboard/test-cafe-cookie-issue/testing/e2e/tests/CookiesTest.test.ts:5:58)



 1/1 failed (12s)
error Command failed with exit code 1.
```

### Screenshots

The URL marked with red should have been proxied, but as can be seen, it is a non proxied url.

<img width="1146" alt="Screen Shot 2022-07-05 at 14 26 47" src="https://user-images.githubusercontent.com/3019331/177416576-b7aa461f-7b61-47dc-b079-a0dd27161262.png">

### Steps to Reproduce

1. Follow the setup steps from the readme.md.
2. Exec `docker-compose up` at repo root
3. Exec `npm run start` at `frontend` dir
4. Exec `npm run start` at `backend` dir

To manually reproduce it.
- Open `local.myapp.app`
- Press the `Set cookies` button
- Press the `Call socket via worker` button.
- You should see `Cookies Found` message, if running outside test cafe.

To run it via test cafe
- Exec `npm run e2e` at `testing` dir
- You should see an error message as a result.

### TestCafe version

"^1.15.1"

### Node.js version

v16.13.2

### Command-line arguments

node ./run-test-cafe.js

### Browser name(s) and version(s)

Chromium 98, Chrome 103

### Platform(s) and version(s)

Mac 12.3

**Comment by Dmitry-Ostashev** [comment]
Hello @kvindascr,
Thank you for your example project. I reproduced the issue.

For the Team:
I've attached a simplified project to see the problem with testcafe-hammerhead playground:

1. `npm install`
2. `node ./index.js`
3. Run testcafe-hammerhead proxy
4. Navigate to the [http://localhost:4000](http://localhost:4000) url. You will see two buttons. Open the DevTools
5. Click the "Send WS message" button. You will see that the web socket connection url looks like ws://localhost:1401/some-random-symbols/[http://localhost:4000](http://localhost:4000)
Click the "Send WS message over new Web Worker" button. You will see that the web socket connection url is not proxied: ws://localhost:4000

**Comment by Artem-Babich** [comment]
Hello,

We've released a new TestCafe version(v2.3.0-rc.2) that [includes the experimental proxyless mode](https://testcafe.io/documentation/404237/guides/experimental-capabilities/proxyless-mode). This mode uses native browser automation. In Proxyless mode, a few issues are already fixed. This issue should also be fixed in Proxyless mode since it should not proxy any URLs, including web sockets. However, I was not able to run your example on my side due to issues that occur on the step with certificates. Would you please check if your sample is working correctly in v2.3.0-rc.2 with the experimental-proxyless mode enabled?
This option is available in all interfaces:

```js
// Command-line
testcafe chrome tests --experimental-proxyless

// Programmatic
const testcafe = await createTestCafe({ experimentalProxyless: true });
    
// Configuration file
{
   "experimentalProxyless": "true"
}    
```

Note that at present it is an experimental mode. Also, the Proxyless mode is implemented only in Google Chrome. It will not work correctly if you run tests in a non-Chrome browser or in a combination of other browsers.
